kyaml/setters2/set.go

```diff
@@ -301,6 +302,33 @@ func validateAgainstSchema(ext *CliExtension, sch *spec.Schema) error {
 	return nil
 }
 
+// fixSchemaTypes traverses the schema and checks for some common
+// errors for the type field. This currently involves users using
+// 'int' instead of 'integer' and 'bool' instead of 'boolean'. Early versions
+// of setters didn't validate this, so there are users that have invalid
+// types in their packages.
+func fixSchemaTypes(sc *spec.Schema) {
```

Comment:
This appears to be a band-aid fix to me. Why can't we advice users to just modify the setters? What if more special cases come in the future from other users ? like list instead of array, max instead of maximum ?

Reply:
Yeah, this is a band-aid. But when we end up accidentally breaking users, we should do what we can to help them out. And we do that by:
 * Fixing the immediate problem (this PR)
 * Making sure it doesn't happen again (thus implementing validation of the schema: https://github.com/kubernetes-sigs/kustomize/pull/2783)
 * Provide a way to migrate the problematic data (We should address this with the fix command).